compressioncodec.go

```diff
@@ -146,3 +282,34 @@ func (c *CompressionSnappyDecoder) Read(p []byte) (int, error) {
 	}
 	return n, err
 }
+
+type CompressionSnappyEncoder struct {
+	w io.Writer
+}
+
+func (c *CompressionSnappyEncoder) Write(p []byte) (int, error) {
+	return 0, fmt.Errorf("Not implemented")
+}
+
+func (c *CompressionSnappyEncoder) Close() error {
+	return fmt.Errorf("Not implemented")
+}
+
+func (c *CompressionSnappyEncoder) Flush() error {
+	return fmt.Errorf("Not implemented")
+}
+
+func compressionHeader(chunkLength int, isOriginal bool) ([]byte, error) {
+	//TODO: Check if this is the right constant
+	if chunkLength > 256*1024 {
```

Comment:
From the docs: "The default compression chunk size is 256K, but writers can choose their own value". I think it would not be correct to assume 256KB here as a writer may want to use a larger value.

Here we are checking that the chunkLength has a reasonable value because the field in the resulting header supports up to 2^23 bytes (~8MB).

After some consideration I think that would be a better upper limit to ensure that no shenanigans happen when we do the conversion.